Status-go crashes when running as daemon with ChainID and NetworkID set to anvil

# Problem

Status-go crashes

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1010bbd98]

goroutine 770 [running]:
github.com/status-im/status-go/services/wallet/connection.(*Status).GetState(0x3733333133000000?)
	/Users/anton/Status_v2/status-go/services/wallet/connection/status.go:55 +0x28
github.com/status-im/status-go/services/wallet/connection.(*Status).SetIsConnected(0x102b5f5c0?, 0x0)
	/Users/anton/Status_v2/status-go/services/wallet/connection/status.go:34 +0x5c
github.com/status-im/status-go/services/wallet/collectibles.(*Manager).checkConnectionStatus(0x140006e4b40, 0x140000dcb40?)
	/Users/anton/Status_v2/status-go/services/wallet/collectibles/manager.go:910 +0x11c
github.com/status-im/status-go/services/wallet/collectibles.(*Manager).FetchAllAssetsByOwner(0x140006e4b40, {0x102f22d58?, 0x14000f069b0?}, 0x7a69, {0xf3, 0x9f, 0xd6, 0xe5, 0x1a, 0xad, ...}, ...)
	/Users/anton/Status_v2/status-go/services/wallet/collectibles/manager.go:259 +0x56c
github.com/status-im/status-go/services/wallet/collectibles.(*Manager).FetchCollectibleOwnershipByOwner(0x4?, {0x102f22d58, 0x14000f069b0}, 0x6?, {0xf3, 0x9f, 0xd6, 0xe5, 0x1a, 0xad, ...}, ...)
	/Users/anton/Status_v2/status-go/services/wallet/collectibles/manager.go:360 +0x50
github.com/status-im/status-go/services/wallet/collectibles.(*loadOwnedCollectiblesCommand).Run(0x14000a985b0, {0x102f22d58, 0x14000f069b0})
	/Users/anton/Status_v2/status-go/services/wallet/collectibles/commands.go:281 +0x404
github.com/status-im/status-go/services/wallet/async.(*Group).Add.func1()
	/Users/anton/Status_v2/status-go/services/wallet/async/async.go:107 +0x3c
created by github.com/status-im/status-go/services/wallet/async.(*Group).Add in goroutine 1090
	/Users/anton/Status_v2/status-go/services/wallet/async/async.go:106 +0x84
```

# Implementation

Known steps towards feature implementation.
What needs further specifying and investigating.

# Acceptance Criteria

Status-go can run as daemon with ChainID and NetworkID set to anvil (31337)

# Notes
config is 
```
{
    "NetworkID": 31337,
    "Networks": [
        {
            "ChainID": 31337,
            "RPCURL": "http://localhost:8545/"
        }
    ],
    "DataDir": "/tmp/status-go-data",
    "NodeKey": "b5c65bcfdb02261052b71cb5f6dec095d0e10df649b4a2b7d881e55e306cbebc",
    "Rendezvous": false,
    "NoDiscovery": false,
    "ListenAddr": "0.0.0.0:30303",
    "HTTPEnabled": true,
    "HTTPHost": "0.0.0.0",
    "HTTPPort": 3333,
    "APIModules": "eth,admin,wallet",
    "WalletConfig": {
      "Enabled": true
    },
    "WakuConfig": {
        "Enabled": false
    }
}
```

command to start 
```
./build/bin/statusd -c ./integration-tests/config.json -log DEBUG --seed-phrase="test test test test test test test test test test test junk" --password="Strong12345"
```
please note, it works fine when ChainID and NetworkID is set to sepolia, e.g. 11155111
# Future Steps

Steps which should be taken after this issue has been resolved.

--- a/services/wallet/collectibles/manager.go
+++ b/services/wallet/collectibles/manager.go
@@ -64,7 +64,7 @@ type Manager struct {
 
 	mediaServer *server.MediaServer
 
-	statuses        map[string]*connection.Status
+	statuses        *sync.Map
 	statusNotifier  *connection.StatusNotifier
 	feed            *event.Feed
 	circuitBreakers sync.Map
@@ -79,26 +79,7 @@ func NewManager(
 	feed *event.Feed) *Manager {
 
 	ownershipDB := NewOwnershipDB(db)
-
-	statuses := make(map[string]*connection.Status)
-
-	allChainIDs := walletCommon.AllChainIDs()
-	for _, chainID := range allChainIDs {
-		status := connection.NewStatus()
-		state := status.GetState()
-		latestUpdateTimestamp, err := ownershipDB.GetLatestOwnershipUpdateTimestamp(chainID)
-		if err == nil {
-			state.LastSuccessAt = latestUpdateTimestamp
-			status.SetState(state)
-		}
-		statuses[chainID.String()] = status
-	}
-
-	statusNotifier := connection.NewStatusNotifier(
-		statuses,
-		EventCollectiblesConnectionStatusChanged,
-		feed,
-	)
+	statuses := initStatuses(ownershipDB)
 
 	return &Manager{
 		rpcClient: rpcClient,
@@ -112,7 +93,7 @@ func NewManager(
 		ownershipDB:        ownershipDB,
 		mediaServer:        mediaServer,
 		statuses:           statuses,
-		statusNotifier:     statusNotifier,
+		statusNotifier:     createStatusNotifier(statuses, feed),
 		feed:               feed,
 	}
 }
@@ -775,7 +756,7 @@ func (o *Manager) fetchCommunityAssets(communityID string, communityAssets []*th
 	return nil
 }
 
-func (o *Manager) fetchCommunityAssetsAsync(ctx context.Context, communityID string, communityAssets []*thirdparty.FullCollectibleData) {
+func (o *Manager) fetchCommunityAssetsAsync(_ context.Context, communityID string, communityAssets []*thirdparty.FullCollectibleData) {
 	if len(communityAssets) == 0 {
 		return
 	}
@@ -807,7 +788,7 @@ func (o *Manager) fillAnimationMediatype(ctx context.Context, asset *thirdparty.
 	return nil
 }
 
-func (o *Manager) processCollectionData(ctx context.Context, collections []thirdparty.CollectionData) error {
+func (o *Manager) processCollectionData(_ context.Context, collections []thirdparty.CollectionData) error {
 	return o.collectionsDataDB.SetData(collections, true)
 }
 
@@ -895,19 +876,33 @@ func (o *Manager) SetCollectibleTransferID(ownerAddress common.Address, id third
 // Reset connection status to trigger notifications
 // on the next status update
 func (o *Manager) ResetConnectionStatus() {
-	for _, status := range o.statuses {
-		status.ResetStateValue()
-	}
+	o.statuses.Range(func(key, value interface{}) bool {
+		value.(*connection.Status).ResetStateValue()
+		return true
+	})
 }
 
 func (o *Manager) checkConnectionStatus(chainID walletCommon.ChainID) {
 	for _, provider := range o.providers.GetProviderList() {
 		if provider.IsChainSupported(chainID) && provider.IsConnected() {
-			o.statuses[chainID.String()].SetIsConnected(true)
+			if status, ok := o.statuses.Load(chainID.String()); ok {
+				status.(*connection.Status).SetIsConnected(true)
+			}
 			return
 		}
 	}
-	o.statuses[chainID.String()].SetIsConnected(false)
+
+	// If no chain in statuses, add it
+	statusVal, ok := o.statuses.Load(chainID.String())
+	status := statusVal.(*connection.Status)
+	if !ok {
+		status = connection.NewStatus()
+		status.SetIsConnected(false)
+		o.statuses.Store(chainID.String(), status)
+		o.updateStatusNotifier()
+	} else {
+		status.SetIsConnected(false)
+	}
 }
 
 func (o *Manager) signalUpdatedCollectiblesData(ids []thirdparty.CollectibleUniqueID) {
@@ -1061,7 +1056,7 @@ func (o *Manager) FetchCollectionSocialsAsync(contractID thirdparty.ContractID) 
 	return nil
 }
 
-func (o *Manager) getOrFetchSocialsForCollection(ctx context.Context, contractID thirdparty.ContractID) (*thirdparty.CollectionSocials, error) {
+func (o *Manager) getOrFetchSocialsForCollection(_ context.Context, contractID thirdparty.ContractID) (*thirdparty.CollectionSocials, error) {
 	socials, err := o.collectionsDataDB.GetSocialsForID(contractID)
 	if err != nil {
 		log.Debug("getOrFetchSocialsForCollection failed for", "chainID", contractID.ChainID, "address", contractID.Address, "err", err)
@@ -1108,4 +1103,32 @@ func (o *Manager) fetchSocialsForCollection(ctx context.Context, contractID thir
 	}
 
 	return socials, cmdRes.Error()
+}
+
+func (o *Manager) updateStatusNotifier() {
+	o.statusNotifier = createStatusNotifier(o.statuses, o.feed)
+}
+
+func initStatuses(ownershipDB *OwnershipDB) *sync.Map {
+	statuses := &sync.Map{}
+	for _, chainID := range walletCommon.AllChainIDs() {
+		status := connection.NewStatus()
+		state := status.GetState()
+		latestUpdateTimestamp, err := ownershipDB.GetLatestOwnershipUpdateTimestamp(chainID)
+		if err == nil {
+			state.LastSuccessAt = latestUpdateTimestamp
+			status.SetState(state)
+		}
+		statuses.Store(chainID.String(), status)
+	}
+
+	return statuses
+}
+
+func createStatusNotifier(statuses *sync.Map, feed *event.Feed) *connection.StatusNotifier {
+	return connection.NewStatusNotifier(
+		statuses,
+		EventCollectiblesConnectionStatusChanged,
+		feed,
+	)
 }

--- a/services/wallet/connection/status_notifier.go
+++ b/services/wallet/connection/status_notifier.go
@@ -2,6 +2,7 @@ package connection
 
 import (
 	"encoding/json"
+	"sync"
 	"time"
 
 	"github.com/ethereum/go-ethereum/common"
@@ -13,21 +14,22 @@ import (
 type StatusNotification map[string]State // id -> State
 
 type StatusNotifier struct {
-	statuses  map[string]*Status // id -> Status
+	statuses  *sync.Map // id -> Status
 	eventType walletevent.EventType
 	feed      *event.Feed
 }
 
-func NewStatusNotifier(statuses map[string]*Status, eventType walletevent.EventType, feed *event.Feed) *StatusNotifier {
+func NewStatusNotifier(statuses *sync.Map, eventType walletevent.EventType, feed *event.Feed) *StatusNotifier {
 	n := StatusNotifier{
 		statuses:  statuses,
 		eventType: eventType,
 		feed:      feed,
 	}
 
-	for _, status := range statuses {
-		status.SetStateChangeCb(n.notify)
-	}
+	statuses.Range(func(_, value interface{}) bool {
+		value.(*Status).SetStateChangeCb(n.notify)
+		return true
+	})
 
 	return &n
 }
@@ -37,13 +39,14 @@ func (n *StatusNotifier) notify(state State) {
 	// a single event, so we fetch them from the map
 	if n.feed != nil {
 		statusMap := make(StatusNotification)
-		for id, status := range n.statuses {
-			state := status.GetState()
+		n.statuses.Range(func(id, value interface{}) bool {
+			state := value.(*Status).GetState()
 			if state.Value == StateValueUnknown {
-				continue
+				return true
 			}
-			statusMap[id] = state
-		}
+			statusMap[id.(string)] = state
+			return true
+		})
 
 		encodedMessage, err := json.Marshal(statusMap)
 		if err != nil {

